pkg/listening-sockets/listening-sockets_test.go

```diff
@@ -0,0 +1,166 @@
+package listeningsockets
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/openshift-kni/commatrix/pkg/client"
+	"github.com/openshift-kni/commatrix/pkg/consts"
+	"github.com/openshift-kni/commatrix/pkg/types"
+	mock_utils "github.com/openshift-kni/commatrix/pkg/utils/mock"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	fakek "k8s.io/client-go/kubernetes/fake"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func normalizeOutput(s string) string {
+	s = strings.ReplaceAll(s, "\t", " ")
+	s = strings.ReplaceAll(s, "\n ", "\n") // Handle newlines with leading spaces
+	s = strings.TrimSpace(s)
+	return s
+}
+
+func TestGenerateSS(t *testing.T) {
+	sch := runtime.NewScheme()
+
+	if err := v1.AddToScheme(sch); err != nil {
+		t.Fatalf("failed to add corev1 to scheme: %v", err)
+	}
+
+	testNode := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-node",
+			Labels: map[string]string{
+				"node-role.kubernetes.io/master": "", // Label to mark this node as a master
+			},
+		},
+	}
+
+	fakeClient := fake.NewClientBuilder().WithScheme(sch).WithObjects(testNode).Build()
+
+	fakeClientset := fakek.NewSimpleClientset()
+
+	clientset := &client.ClientSet{
+		Client:          fakeClient,
+		CoreV1Interface: fakeClientset.CoreV1(),
+	}
+
+	ctrlTest := gomock.NewController(t)
+	defer ctrlTest.Finish()
+
+	mockUtils := mock_utils.NewMockUtilsInterface(ctrlTest)
+
+	tcpOutput := []byte(`LISTEN 0      4096      127.0.0.1:8797  0.0.0.0:* users:(("machine-config-",pid=3534,fd=3))                
+LISTEN 0      4096      127.0.0.1:8798  0.0.0.0:* users:(("machine-config-",pid=3534,fd=13))               
+LISTEN 0      4096      127.0.0.1:9100  0.0.0.0:* users:(("node_exporter",pid=4147,fd=3))`)
+
+	udpOutput := []byte(`UNCONN 0      0           0.0.0.0:111   0.0.0.0:* users:(("rpcbind",pid=1399,fd=5),("systemd",pid=1,fd=78))
+UNCONN 0      0         127.0.0.1:323   0.0.0.0:* users:(("chronyd",pid=1015,fd=5))                        
+UNCONN 0      0      10.46.97.104:500   0.0.0.0:* users:(("pluto",pid=2115,fd=21))`)
+
+	output := []byte(`1: /system.slice/crio-1234567890abcdef.scope
+2: /system.slice/other-service.scope
+3: /system.slice/sshd.service`)
+
+	mockUtils.EXPECT().RunCommandOnPod(gomock.Any(), gomock.Any()).DoAndReturn(
+		func(pod *v1.Pod, cmd []string) ([]byte, error) {
+			if len(cmd) > 2 && strings.HasPrefix(cmd[2], "crictl ps -o json --id") {
+				return []byte(`{
+					"containers": [
+						{
+							"labels": {
+								"io.kubernetes.container.name": "test-container",
+								"io.kubernetes.pod.name": "test-pod",
+								"io.kubernetes.pod.namespace": "test-namespace"
+							}
+						}
+					]
+				}`), nil
+			}
+			if strings.HasPrefix(cmd[2], "cat /proc/") && strings.Contains(cmd[2], "/cgroup") {
+				return output, nil
+			}
+			if cmd[2] == "ss -anpltH" {
```

Comment:
You can avoid these ifs by replacing these commands like in the next expects

Reply:
Because  its from same function that we want to mock maybe better to have it like this?

Reply:
If you can have them in a one liner instead of 5 ifs I think it makes the code more readable